ui/src/layouts/storage/functions.rs

```diff
@@ -336,6 +336,8 @@ pub fn init_coroutine<'a>(
                         file_name,
                         local_path_to_save_file,
                     } => {
+                        let (local_path_to_save_file, on_finish) =
+                            get_download_path(local_path_to_save_file);
```

Comment:
since this is in a async function, it would be best to use async-like functions for handling this to prevent any possible blocking of the current thread its operating in.

Reply:
you mean the on finished part?

Reply:
Correct